rust/nasl-builtin-cryptographic/src/hash.rs

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+use digest::Digest;
+use md2::Md2;
+use md4::Md4;
+use md5::Md5;
+use nasl_builtin_utils::error::FunctionErrorKind;
+use ripemd::Ripemd160;
+use sha1::Sha1;
+use sha2::{Sha256, Sha512};
+
+use crate::NaslFunction;
+use nasl_builtin_utils::{Context, Register};
+use nasl_syntax::NaslValue;
+
+fn nasl_hash<D: Digest>(register: &Register) -> Result<NaslValue, FunctionErrorKind>
+where
+    D::OutputSize: std::ops::Add,
+    <D::OutputSize as std::ops::Add>::Output: digest::generic_array::ArrayLength<u8>,
+{
+    let positional = register.positional();
+    if positional.is_empty() {
+        return Err(FunctionErrorKind::MissingPositionalArguments {
+            expected: 0,
+            got: 1,
+        });
+    };
+    let data = match &positional[0] {
+        NaslValue::String(x) => x,
+        NaslValue::Null => return Ok(NaslValue::Null),
```

Comment:
Not sure, if I would allow in a null value in a hash function

Reply:
This is how c nasl works. I can not change the current behavior and can't stop the nasl execution. It is up to the nasl developer if they stop or not in case they get a null response.